### PR TITLE
Add missing code-style rules

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #28791.

Adds:

- [IDE0150](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0150?branch=pr-en-us-30059)
- [IDE0170](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0170?branch=pr-en-us-30059)
- [IDE0180](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0180?branch=pr-en-us-30059)

Renames option for IDE0039.

### CODE REVIEW

File: docs/fundamentals/code-analysis/style-rules/ide0150.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: "IDE0150: Prefer 'null' check over type check"
+description: "Learn about code analysis rule IDE0150: Prefer 'null' check over type check"
+ms.date: 06/29/2022
+ms.topic: reference
+f1_keywords:
+- IDE0150
+helpviewer_keywords:
+- IDE0150
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0150.md:9]:
```suggestion
f1_keywords:
- IDE0150
- csharp_style_prefer_null_check_over_type_check
helpviewer_keywords:
- IDE0150
- csharp_style_prefer_null_check_over_type_check
```

File: docs/fundamentals/code-analysis/style-rules/ide0170.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+---
+title: "IDE0170: Simplify property pattern"
+description: "Learn about code analysis rule IDE0170: Simplify property pattern"
+ms.date: 06/29/2022
+ms.topic: reference
+f1_keywords:
+- IDE0170
+helpviewer_keywords:
+- IDE0170
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0170.md:9]:
```suggestion
f1_keywords:
- IDE0170
- csharp_style_prefer_extended_property_pattern
helpviewer_keywords:
- IDE0170
- csharp_style_prefer_extended_property_pattern
```

File: docs/fundamentals/code-analysis/style-rules/ide0180.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+---
+title: "IDE0180: Use tuple to swap values"
+description: "Learn about code analysis rule IDE0180: Use tuple to swap values"
+ms.date: 06/29/2022
+ms.topic: reference
+f1_keywords:
+- IDE0180
+helpviewer_keywords:
+- IDE0180
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/ide0180.md:9]:
```suggestion
f1_keywords:
- IDE0180
- csharp_style_prefer_tuple_swap
helpviewer_keywords:
- IDE0180
- csharp_style_prefer_tuple_swap
```

**Review by Youssef1313 — APPROVED** [review]

File: docs/fundamentals/code-analysis/style-rules/ide0150.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
+---
+title: "IDE0150: Prefer 'null' check over type check"
+description: "Learn about code analysis rule IDE0150: Prefer 'null' check over type check"
+ms.date: 06/29/2022
+ms.topic: reference
+f1_keywords:
+- IDE0150
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0150.md:7]:
```suggestion
- IDE0150
- csharp_style_prefer_null_check_over_type_check
```

File: docs/fundamentals/code-analysis/style-rules/ide0150.md (hunk 1 above)
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0150.md:9]:
```suggestion
- IDE0150
- csharp_style_prefer_null_check_over_type_check
```

**Review by BillWagner — APPROVED** [review]
Thanks again @gewarren 

Once other comments are resolved, this is ready to :shipit:

**Comment by gewarren** [comment]
Thanks Bill and Youssef!

**Pull request merged**